backend/shop/tests/test_views.py

```diff
@@ -487,3 +490,118 @@ def test_get_shop_profiles_success(self):
 
         # Assert that the returned groups match the expected groups
         self.assertEqual(set(returned_groups), set(expected_groups))
+
+
+class TestProductGet(APITestCase):
+    def setUp(self):
+        self.user = CoreUserFactory()
+        self.url_all = reverse("product_get_all")
+
+    def test_product_get_success(self):
+        perm_view_product = Permission.objects.get(codename="view_product")
+        self.user.user_permissions.add(perm_view_product)
+        self.client.force_authenticate(user=self.user)
+
+        product = ProductFactory()
+        response = self.client.get(self.url_all)
+        self.assertEqual(response.status_code, 200)
+
+        # check if returned data is product
+        expected_keys = {"id", "image", "title", "description", "price"}
+        self.assertTrue(expected_keys.issubset(response.data[0].keys()))
+
+    def test_product_get_success_permission_change_product(self):
+        perm_view_product = Permission.objects.get(codename="change_product")
+        self.user.user_permissions.add(perm_view_product)
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.get(self.url_all)
+        self.assertEqual(response.status_code, 200)
+
+    def test_product_get_failure_not_authenticated(self):
+        response = self.client.get(self.url_all)
+        self.assertEqual(response.status_code, 302)
+
+    def test_product_get_failure_no_permissions(self):
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.get(self.url_all)
+        self.assertEqual(response.status_code, 403)
+
+    def test_product_get_one_success_using_product_id(self):
+        perm_view_product = Permission.objects.get(codename="view_product")
+        self.user.user_permissions.add(perm_view_product)
+        self.client.force_authenticate(user=self.user)
+
+        product = ProductFactory(title="test_title")
+        url = reverse("product_get", kwargs={"product_id": product.id})
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(response.data["title"], "test_title")
+
+
+class TestProductCreate(APITestCase):
```

Comment:
also here, your manipulation of permissions is just noise in this test, this needs to a separate concern that is part of `setUp`